src/ray/common/scheduling/cluster_resource_scheduler.h

```diff
@@ -69,8 +69,18 @@ class TaskResourceInstances {
   /// The list of instances of each custom resource allocated to a task.
   absl::flat_hash_map<int64_t, std::vector<double>> custom_resources;
   bool operator==(const TaskResourceInstances &other);
+  /// For each resource of this request aggregate its instances.
+  TaskRequest ToTaskRequest();
```

Comment:
I noticed that the `TaskRequest` class doesn't have a comment explaining what it is and how it relates to `TaskResourceInstances`. Can you add one?

By the way, `TaskRequest` looks like it could be a struct with all `const` members (right now it's a `class` with all non-const members).